src/main/java/com/google/opensesame/projects/ProjectQuery.java

```diff
@@ -0,0 +1,143 @@
+   * @return Returns the list of QueryFilters.
+   * @throws IOException
+   * @throws ServletValidationException Throws if any of the query filter strings in the request are
+   *     invalid. See {@link com.google.opensesame.projects.QueryFilter.fromString} for more
+   *     information on how QueryFilters are parsed.
+   */
+  private static List<QueryFilter> getQueryFiltersFromRequest(HttpServletRequest request)
+      throws IOException, ServletValidationException {
+    String[] filterRequests = request.getParameterValues(FILTER_QUERY_PARAM);
+    if (filterRequests == null) {
+      return Arrays.asList();
+    }
+
+    List<QueryFilter> queryFilters = new ArrayList<QueryFilter>();
+    for (String filterRequest : filterRequests) {
+      QueryFilter queryFilter = QueryFilter.fromString(filterRequest);
+      queryFilters.add(queryFilter);
+    }
+
+    return queryFilters;
+  }
+
+  /**
+   * Queries for ProjectEntities in the datastore based on a servlet request.
+   *
+   * @param request The servlet request defining the query.
+   * @return Returns a collection of ProjectEntities from the query.
+   * @throws IOException
+   * @throws ServletValidationException Throws if the query request is invalid.
+   */
+  public static Collection<ProjectEntity> queryFromRequest(HttpServletRequest request)
+      throws IOException, ServletValidationException {
+    String[] projectIds = request.getParameterValues(ProjectEntity.PROJECT_ID_PARAM);
+
+    if (projectIds != null) {
+      return getProjectEntitiesByIds(projectIds);
+    } else {
+      // TODO(Richie): Add pagination support.
+      // TODO(Richie): Add ordering support.
+      List<QueryFilter> queryFilters = getQueryFiltersFromRequest(request);
+      return getProjectEntitiesByQueryFilters(queryFilters);
+    }
+  }
+
+  /**
+   * Gets the ProjectEntities in the intersection of a list of QueryFilters.
+   *
+   * @param queryFilters The list of QueryFilters.
+   * @return Returns a collection of ProjectEntities in the intersection of the list of
+   *     QueryFilters.
+   */
+  public static Collection<ProjectEntity> getProjectEntitiesByQueryFilters(
+      List<QueryFilter> queryFilters) {
+    if (queryFilters.size() == 0) {
+      return ofy().load().type(ProjectEntity.class).list();
+    }
+
+    List<ProjectEntity> intermediateIntersection = null;
+    for (QueryFilter queryFilter : queryFilters) {
+      List<ProjectEntity> filteredProjects =
+          ofy()
+              .load()
+              .type(ProjectEntity.class)
+              .filter(queryFilter.condition, queryFilter.comparisonObject)
+              .list();
+
+      if (intermediateIntersection != null) {
+        intermediateIntersection =
+            filteredProjects.stream()
+                .filter(intermediateIntersection::contains)
+                .collect(Collectors.toList());
+      } else {
+        intermediateIntersection = filteredProjects;
+      }
+    }
```

Comment:
There should be a way to apply multiple filters at once. I'm not super familiar with what is going on but would something like this work?
```suggestion
    List<ProjectEntity> filteredProjects =
          ofy()
              .load()
              .type(ProjectEntity.class)
    for (QueryFilter queryFilter : queryFilters) {
              filteredProjects = filteredProjects.filter(queryFilter.condition, queryFilter.comparisonObject)
    }
    return filteredProjects.list()

Reply:
I wish! This is very similar to the code that I had originally written for this functionality, but due to how Datastore works internally you can only make one filter per query to the Datastore. That's why the best option with Datastore is to make multiple queries and determine the intersection of all of the filters. It's not amazingly efficient, but if this ever becomes a problem, we can make use of [projection queries](https://cloud.google.com/appengine/docs/standard/java/datastore/projectionqueries) to pull limited data from Datastore on the intermediate queries.